### PR TITLE
feat: extend write_deltalake to accept Deltalake schema

### LINKED ISSUE #1862: Unable to use deltalake `Schema` in `write_deltalake`
# Environment

**Delta-rs version**: 0.13.0
**Binding**: Python 3.11
**OS**: Windows 10 WSL2 (Ubuntu 22.04.02 LTS)

***
# Bug

I want to use a `deltalake.schema.Schema` instance in the function `write_deltalake(schema=...)` in order to control the nullable parameter and add metadata. However,  a ` pyarrow.lib.Schema` type is expected instead of `deltalake._internal.Schema`. This is issue is reproducible with Polars and Pandas as shown with the snippet below

```python
import json

import polars as pl
import pandas as pd
from deltalake import write_deltalake
from deltalake.schema import Schema


schema = Schema.from_json(
    json.dumps(
        {
            "type": "struct",
            "fields": [
                {"name": "x", "type": "integer", "nullable": True, "metadata": {"description": "variable x"}},
                {"name": "y", "type": "float", "nullable": False, "metadata": {"description": "variable y"}},
            ],
        }
    )
)

data = {"x": [1, 2], "y": [3.0, 4.0]}

try:
    write_deltalake(
        "data/test_schema",
        data=pl.DataFrame(data),
        name="test",
        schema=schema,
        description="A test",
        mode="overwrite",
    )
except TypeError as e:
    print(f"Polars -> {e}")

try:
    write_deltalake(
        "data/test_schema",
        data=pd.DataFrame(data),
        name="test",
        schema=schema,
        description="A test",
        mode="overwrite",
    )
except TypeError as e:
    print(f"Pandas -> {e}")
```

```bash
Polars -> Argument 'schema' has incorrect type (expected pyarrow.lib.Schema, got deltalake._internal.Schema)
Pandas -> Argument 'schema' has incorrect type (expected pyarrow.lib.Schema, got deltalake._internal.Schema)
```

### PR DESCRIPTION
**Pull request by r3stl355**
# Description
Extended write_deltalake to accept either PyArrow or Deltalake schema. 
Added a test

# Related Issue(s)
closes  #1862

### CODE REVIEW

**Comment by r3stl355** [comment]
interesting, caught by type check in Python :)

File: python/deltalake/writer.py
Comment (ion-elgreco) [inline python/deltalake/writer.py:184]: This DeltaLake.schema to pyarrow schema needs happen also when we do `pa.Table.from_pandas()`.

Comment (r3stl355) [reply python/deltalake/writer.py:184]: good point, missed that, will fix

Comment (r3stl355) [reply python/deltalake/writer.py]: Agree, makes sense to wait until that PR is merged as it changes the same file. I'll refresh the PR once the other one is merged

**Review by wjones127 — APPROVED** [review]
Thanks for working on this!

**Comment by r3stl355** [comment]
closing this as something got messed up with rebase, will create a different PR

**Pull request closed without merging**
